spec/math_spec.rb

```diff
@@ -377,6 +376,33 @@
                    1.4000e-03, 1.0800e-03, 3.4400e-04], dtype: dtype)
                 )
       end
+    end unless rational_dtype?(dtype) or integer_dtype?(dtype)
+
+    context "#pow #{dtype}" do
+      next if integer_dtype?(dtype)
+
+      before do 
+        @n = NMatrix.new [4,4], [0, 2, 0, 1,
+                                2, 2, 3, 2,
+                                4,-3, 0, 1,
+                                6, 1,-6,-5], dtype: dtype
+      end
+
+      it "raises a square matrix to a even power" do
+        expect(@n.pow(4)).to eq(NMatrix.new([4,4], [292, 28,-63, -42, 
+                                                   360, 96, 51, -14,
+                                                   448,-231,-24,-87,
+                                                 -1168, 595,234, 523], 
+                                                 dtype: dtype))
+      end
+
+      it "raises a square matrix to odd power" do
+        expect(@n.pow(9)).to eq(NMatrix.new([4,4],[-275128,  279917, 176127, 237451,
+                                                  -260104,  394759, 166893,  296081,
+                                                  -704824,  285700, 186411,  262002,
+                                                  3209256,-1070870,-918741,-1318584],
+                                                  dtype: dtype))
+      end
```

Comment:
You're supporting negative powers and all stypes, so you need tests for negative powers and all combinations of stype/dtype.